src/mute/mutedUsersReducer.js

```diff
@@ -0,0 +1,35 @@
+/* @flow strict-local */
+import Immutable from 'immutable';
+import type { MutedUsersState, Action, UserId } from '../types';
+import { REALM_INIT, LOGOUT, ACCOUNT_SWITCH, EVENT_MUTED_USERS } from '../actionConstants';
+import { makeUserId } from '../api/idTypes';
+import type { MutedUser } from '../api/apiTypes';
+
+const initialState: MutedUsersState = Immutable.Map({});
+
+function mutedUsersToMap(muted_users: MutedUser[]): Immutable.Map<UserId, number> {
+  if (muted_users) {
+    return Immutable.Map(
+      muted_users.map(muted_user => [makeUserId(muted_user.id), muted_user.timestamp]),
+    );
+  } else {
+    return initialState;
+  }
```

Comment:
This was guarding against it being undefined for the javascript `.map`, but I've switched to the cleaner way you suggested in a later comment. Done.

Reply:
Ah, that makes sense. In fact I think I was forgetting that an empty JS array is truthy (forgetting because an empty Python list is falsy).

In a situation like this but where `?? []` wasn't a handy solution so that we did need some form of conditional like this, I'd probably do it with the early-return pattern:
```js
  if (!muted_users) {
    return initialState;
  }
  return Immutable.Map(muted_users.map(/* … */));
```

Oh, also in that case I'd definitely want the function to reflect in its signature that the thing could be undefined :slightly_smiling_face: . So `muted_users: MutedUser[] | void`, or perhaps `muted_users?: MutedUser[]`. That way the type-checker sees what I'm doing with the check, and future people touching the code won't accidentally refactor away the check without noticing.